infrastructure/restapi/src/testFixtures/java/tech/pegasys/teku/infrastructure/restapi/StubRestApiRequest.java

```diff
@@ -120,11 +120,11 @@ private void setOptionalQueryParameter(final String parameter, final String valu
   }
 
   @Override
-  @SuppressWarnings("unchecked")
   public <T> T getPathParameter(final ParameterMetadata<T> parameterMetadata) {
     assertThat(this.pathParameters.containsKey(parameterMetadata.getName())).isTrue();
-    final Object param = pathParameters.get(parameterMetadata.getName());
-    return (T) param;
+    return parameterMetadata
```

Comment:
previously handled strings fine, which covers beacon-api, but the new parameter results in a BlsPublicKey directly so this fn had to get enhanced to use the deserializer